### PR TITLE
Fix #3583: Merge topic_practice_footer_view into single layout file.

### LINKED ISSUE #3583: Merge topic_practice_footer_view.xml into single xml file
Currently there are 4 versions of `topic_practice_footer_view.xml` file merge it into single xml file.

We can use https://text-compare.com/ to compare two versions of this file and for all the differences we can create variables in `dimens.xml` file and use it accordingly.

**Note**: In PR, make sure you add before and after screenshot of mobile-portrait, mobile-landscape, tablet-portrait and tablet-landscape for comparison and make sure that there is not difference between before and after UI.

### PR DESCRIPTION
**Pull request by rishidyno**
## Explanation

 Fix #3583

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

| OLD Screenshots         | NEW Screenshots |
| -----------       | -----------      |
|Screenshot_20210804_152357 | Screenshot_20210804_152357 |
| Screenshot_20210804_152412 | Screenshot_20210804_152412 |
| Screenshot_20210804_152456 |  Screenshot_20210804_152456|
| Screenshot_20210804_152510 | Screenshot_20210804_152510|

### CODE REVIEW

**Review by rt4914 — CHANGES_REQUESTED** [review]
@rishidyno There are 4 files which needs to be merged into 1.

Also as mentioned in issue please add before/after screenshots of this UI in app in mobile portrait/landscape and tablet portrait/landscape mode.

**Comment by rishidyno** [comment]
> @rishidyno There are 4 files which needs to be merged into 1.
> 
> Also as mentioned in issue please add before/after screenshots of this UI in app in mobile portrait/landscape and tablet portrait/landscape mode.

@rt4914  I have added Screenshots also I wanted to mention that I tried merging it into one XML file but I ran into some issues.

I added these two in  `mobile portrait/landscape` `dimens.xml` beacause these two are identicle.

  `<dimen name="topic_practice_footer_view_margin_top">36dp</dimen>`
  `<dimen name="topic_practice_footer_view_margin_bottom">108dp</dimen>`

Also i added these two in `tablet portrait/landscape` `dimens.xml` beacause these two are identicle.

  `<dimen name="topic_practice_footer_view_margin_top">24dp</dimen>`
  `<dimen name="topic_practice_footer_view_margin_bottom">0dp</dimen>`

But when I tried to run the app in that activity it crashed in tablet portrait/landscape so I kept the tablet portrait and landscape file.

I could not get a complete understanding of why they crashed. But I think it might be due to other layout elements surrounding them.

I could only delete the landscape file. 

Please guide me on what to do now.

**Comment by rt4914** [comment]
> > @rishidyno There are 4 files which needs to be merged into 1.
> > Also as mentioned in issue please add before/after screenshots of this UI in app in mobile portrait/landscape and tablet portrait/landscape mode.
> 
> @rt4914 I have added Screenshots also I wanted to mention that I tried merging it into one XML file but I ran into some issues.
> 
> I added these two in `mobile portrait/landscape` `dimens.xml` beacause these two are identicle.
> 
> `<dimen name="topic_practice_footer_view_margin_top">36dp</dimen>`
> `<dimen name="topic_practice_footer_view_margin_bottom">108dp</dimen>`
> 
> Also i added these two in `tablet portrait/landscape` `dimens.xml` beacause these two are identicle.
> 
> `<dimen name="topic_practice_footer_view_margin_top">24dp</dimen>`
> `<dimen name="topic_practice_footer_view_margin_bottom">0dp</dimen>`
> 
> But when I tried to run the app in that activity it crashed in tablet portrait/landscape so I kept the tablet portrait and landscape file.
> 
> I could not get a complete understanding of why they crashed. But I think it might be due to other layout elements surrounding them.
> 
> I could only delete the landscape file.
> 
> Please guide me on what to do now.

@rishidyno After you have done the changes for by removing the tablet xml file also, rebuild the entire project, uninstall the previous app from tablet device and then install the app. It should work properly after that.

**Comment by rishidyno** [comment]
> > > @rishidyno There are 4 files which needs to be merged into 1.
> > > Also as mentioned in issue please add before/after screenshots of this UI in app in mobile portrait/landscape and tablet portrait/landscape mode.
> > 
> > 
> > @rt4914 I have added Screenshots also I wanted to mention that I tried merging it into one XML file but I ran into some issues.
> > I added these two in `mobile portrait/landscape` `dimens.xml` beacause these two are identicle.
> > `<dimen name="topic_practice_footer_view_margin_top">36dp</dimen>`
> > `<dimen name="topic_practice_footer_view_margin_bottom">108dp</dimen>`
> > Also i added these two in `tablet portrait/landscape` `dimens.xml` beacause these two are identicle.
> > `<dimen name="topic_practice_footer_view_margin_top">24dp</dimen>`
> > `<dimen name="topic_practice_footer_view_margin_bottom">0dp</dimen>`
> > But when I tried to run the app in that activity it crashed in tablet portrait/landscape so I kept the tablet portrait and landscape file.
> > I could not get a complete understanding of why they crashed. But I think it might be due to other layout elements surrounding them.
> > I could only delete the landscape file.
> > Please guide me on what to do now.
> 
> @rishidyno After you have done the changes for by removing the tablet xml file also, rebuild the entire project, uninstall the previous app from tablet device and then install the app. It should work properly after that.

Thanks a lot @rt4914 I followed what you said and I was able to merge it into a single layout file. 
Now it only needs to be reviewed by you.

File: .idea/misc.xml [hunk 1]
```diff
@@ -1,9 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<project version="4">
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
-    <output url="file://$PROJECT_DIR$/build/classes" />
-  </component>
-  <component name="ProjectType">
```
Comment (prayutsu) [inline .idea/misc.xml]: You only needed to revert the unwanted changes and not delete the file, please restore this file.

File: .idea/runConfigurations.xml [hunk 2]
```diff
@@ -1,12 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<project version="4">
-  <component name="RunConfigurationProducerService">
-    <option name="ignoredProducers">
-      <set>
-        <option value="org.jetbrains.plugins.gradle.execution.test.runner.AllInPackageGradleConfigurationProducer" />
-        <option value="org.jetbrains.plugins.gradle.execution.test.runner.TestClassGradleConfigurationProducer" />
-        <option value="org.jetbrains.plugins.gradle.execution.test.runner.TestMethodGradleConfigurationProducer" />
-      </set>
```
Comment (prayutsu) [inline .idea/runConfigurations.xml]: Ditto

**Review by prayutsu — CHANGES_REQUESTED** [review]
Left two comments, PTAL @rishidyno.
Also, note that if you want any reviewers to add their review, you should assign them the PR. Reviewers only add their reviews to the PRs assigned to them.

File: .idea/misc.xml (hunk 1 above)
Comment (rishidyno) [reply .idea/misc.xml]: I have restored this file

File: .idea/runConfigurations.xml (hunk 2 above)
Comment (rishidyno) [reply .idea/runConfigurations.xml]: and this as well

**Comment by rishidyno** [comment]
Ptal @rt4914 

> Left two comments, PTAL @rishidyno.
> Also, note that if you want any reviewers to add their review, you should assign them the PR. Reviewers only add their reviews to the PRs assigned to them.

Got it 🙂

**Review by prayutsu — APPROVED** [review]
LGTM, Thanks @rishidyno

**Comment by rishidyno** [comment]
> LGTM, Thanks @rishidyno

Thanks @prayutsu

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 3]
```diff
@@ -290,4 +290,8 @@
   <dimen name="coming_soon_topic_promoted_list_view_padding_start">0dp</dimen>
   <dimen name="coming_soon_topic_Promoted_list_view_padding_end">0dp</dimen>
 
+  <!--  Topic practice footer view-->
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: Nit spacing missing
`<!--  Topic practice footer view -->`

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 4]
```diff
@@ -293,4 +293,8 @@
   <dimen name="coming_soon_topic_promoted_list_view_padding_start">0dp</dimen>
   <dimen name="coming_soon_topic_Promoted_list_view_padding_end">0dp</dimen>
 
+  <!--  Topic practice footer view-->
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-port/dimens.xml]: Nit spacing missing
`<!--  Topic practice footer view -->`

File: app/src/main/res/values/dimens.xml [hunk 5]
```diff
@@ -450,4 +450,8 @@
   <dimen name="coming_soon_topic_promoted_list_view_padding_start">28dp</dimen>
   <dimen name="coming_soon_topic_Promoted_list_view_padding_end">28dp</dimen>
 
+  <!--  Topic practice footer view-->
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml]: Nit spacing missing
`<!--  Topic practice footer view -->`

**Review by rt4914 — CHANGES_REQUESTED** [review]
@rishidyno Code is good. Just nit changes suggested.

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 3 above)
Comment (rishidyno) [reply app/src/main/res/values-sw600dp-land/dimens.xml]: @rt4914  I have added spaces and capitalized this.

File: app/src/main/res/values-sw600dp-port/dimens.xml (hunk 4 above)
Comment (rishidyno) [reply app/src/main/res/values-sw600dp-port/dimens.xml]: same

File: app/src/main/res/values/dimens.xml (hunk 5 above)
Comment (rishidyno) [reply app/src/main/res/values/dimens.xml]: done

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**